tests/core_tests/block_reward.cpp

```diff
@@ -172,21 +172,21 @@ bool gen_block_reward::generate(std::vector<test_event_entry>& events) const
     return false;
 
   // Test: fee increases block reward
-  transaction tx_0(construct_tx_with_fee(events, blk_5, miner_account, bob_account, MK_COINS(1), 3 * TESTS_DEFAULT_FEE));
+  transaction tx_0(construct_tx_with_fee(events, blk_5r, miner_account, bob_account, MK_COINS(1), 3 * TESTS_DEFAULT_FEE));
```

Comment:
Can you give a rationale of changing `blk_5` to `blk_5r` here, and for similar other changes in the tests? This code has never been changed before, so for sure the current Trezor changes shouldn't require it. If it's some bug in tests, I'd like to know where it was and why it required such changes.

Reply:
> monero tests: fix tx create from sources, input locking. Originally, creating a synthetic transactions with chaingen could create a transaction with outputs that are still locked in the current block, thus failing chain validation by the daemon. Simple unlock check was added. Some buggy tests were fixed as well as new unlock-checking version of tx creation rejected those, fixes are simple - mostly using correct block after a rewind to construct a transaction

I've compiled this PR and I'm playing with tests now, to look how this all works because I didn't touch this code before.

Reply:
Thanks for starting the review! ❤️ 
 Regarding the blk_5r question - I had to rewind so tests pass after fix of tx lock time.

Reply:
@SChernykh can I pls resolve this one?